packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol

```diff
@@ -200,6 +213,21 @@ contract WorldRegistrationSystem is System, IWorldErrors {
     return worldFunctionSelector;
   }
 
+  /**
+   * Register root World function selectors in batch (without namespace / name prefix).
+   * Requires the caller to own the root namespace.
+   */
+  function registerRootFunctionSelectors(
+    bytes32 resourceSelector,
+    bytes4[] calldata worldFunctionSelectors,
+    bytes4[] calldata systemFunctionSelectors
+  ) public {
+    // Register the function selectors
+    for (uint256 i = 0; i < worldFunctionSelectors.length; i++) {
+      registerRootFunctionSelector(resourceSelector, worldFunctionSelectors[i], systemFunctionSelectors[i]);
```

Comment:
do we need to validate that `worldFunctionSelectors.length == systemFunctionSelectors.length`?